arch/arm/src/stm32h5/stm32_pwm.c

```diff
@@ -0,0 +1,4428 @@
+/****************************************************************************
+ * arch/arm/src/stm32h5/stm32_pwm.c
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <assert.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <nuttx/arch.h>
+#include <arch/board/board.h>
+
+#include "arm_internal.h"
+#include "chip.h"
+#include "stm32_pwm.h"
+#include "stm32.h"
+
+/* This module then only compiles if there is at least one enabled timer
+ * intended for use with the PWM upper half driver.
+ *
+ * It implements support for both:
+ *   1. STM32 TIMER IP version 1 - F0, F1, F2, F37x, F4, L0, L1
+ *   2. STM32 TIMER IP version 2 - F3 (no F37x), F7, H7, L4, L4+
+ */
+
+#ifdef CONFIG_STM32H5_PWM
```

Comment:
You probably don't need this guard if the file doesn't get included in the build by the Makefile (you have a guard there).